### ISSUE TITLE
Setting `strict` from Config instead of CLI

### ISSUE THREAD

**Issue by rrrene** [rrrene/credo#214]
This continues a discussion from #207.

I added a function to manage the (implicit) settings of `strict` mode in eb546b9.

This should make setting `strict` more explicit while keeping the internal consequences more implicit.

Toughts?

/cc @sstern6 @devonestes

**Comment by sstern6** [comment]
Thanks for continuing the thread @rrrene.

Thanks for reworking this, I like what you've done IMO.

Not sure how much more "isolated", this needs to be with respect to `cli.ex` vs `config.ex` at the moment.

What is concerning to me is the lack of tests for `--strict` and `strict: true`, incase a PR is merged that effects the setting of strict internally, which if released can cause a lot of issues.  Is there a way you can recommend stubbing/testing the config/cli values of strict?  Do you think this is needed?  If so and you can point me in a direction I can write some tests for the cli and config strict.

Final thought(which might considering to many edge cases, but wanted to throw it out there), do we want to restrict users in the config to only setting `true/false`, or should we allow any interpretation of what the user wants to for the boolean outcome, for example, allowing `!false`, `:true`, etc?

Thanks

**Comment by devonestes** [comment]
Also, the thing that I thought about briefly while working on this feature was the following idea:

So, ideally, we want anything passed as a command line flag to be able to override any settings in `.credo.exs`, right? Well, right now if we have `strict: false` in the config, then we can override that with `--strict`. But what if there's `strict: true` in the config and we want to override that with something on the command line (like maybe `--no-strict`). I'm not 100% sure that this is a valid concern, but since we're talking about this stuff I figured I'd actually raise the point.

In general I try not to add features that haven't been asked for, but I figured I'd toss it out there to see what you all think.

Also, I'd be happy to write some tests as well - just point me in the right direction as to what you want done!
